src/main/java/discord4j/discordjson/json/ApplicationCommandInteractionOptionData.java

```diff
@@ -0,0 +1,33 @@
+package discord4j.discordjson.json;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import discord4j.discordjson.possible.Possible;
+import org.immutables.value.Value;
+
+import java.util.List;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableApplicationCommandInteractionOptionData.class)
+@JsonDeserialize(as = ImmutableApplicationCommandInteractionOptionData.class)
+public interface ApplicationCommandInteractionOptionData {
+
+    static ImmutableApplicationCommandInteractionOptionData.Builder builder() {
+        return ImmutableApplicationCommandInteractionOptionData.builder();
+    }
+
+    /**
+     * the name of the parameter
+     */
+    String name();
+
+    /**
+     * the value of the pair
+     */
+    Possible<String> value();
```

Comment:
Isn't it an int? https://discord.com/developers/docs/interactions/slash-commands#applicationcommandoptiontype

Reply:
Not quite; value is of *type* OptionType, which allows receiving a value of type user, channel, string, etc. I've made this a String here as it's a supertype of all the serialised values.

Reply:
Oh okay, so what are the values (int) in the doc I linked?

Reply:
Thanks for the explanation, I now understand